src/core_plugins/tagcloud/public/tag_cloud_visualization.js

```diff
@@ -92,7 +92,11 @@ export class TagCloudVisualization {
     }
 
     const data = response.tables[0];
-    this._bucketAgg = this._vis.aggs.find(agg => agg.type.name === 'terms');
+    if (this._vis.aggs.bySchemaName.segment) {
+      this._bucketAgg = this._vis.aggs.bySchemaName.segment[0];
```

Comment:
this might throw, if segment doesn't have any elements ... maybe use lodash ?
`this._bucketAgg = _.get(this, '_vis.aggs.bySchemaName.segment[0]', null)`